src/utils/matrix.ts

```diff
@@ -270,11 +270,17 @@ export async function createRoom(
 
   // Wait for the room to arrive
   await new Promise<void>((resolve, reject) => {
-    const onRoom = async (room: Room): Promise<void> => {
-      if (room.roomId === (await createPromise).room_id) {
-        resolve();
-        cleanUp();
-      }
+    const onRoom = (room: Room): void => {
+      createPromise
+        .then((result) => {
+          if (room.roomId === result.room_id) {
+            resolve();
+            cleanUp();
+          }
+        })
+        .catch((e) => {
+          logger.error("Failed to wait for the room to arrive", e);
+        });
     };
```

Comment:
Since the await on the inlined promise requires `createPromise` to settle, we can avoid having to await on `createPromise` again to pull the result out, by having the inlined promise resolve to that result:

```typescript
  const roomId = await new Promise<string>((resolve, reject) => {
    const onRoom = (room: Room): void => {
      createPromise
        .then((result) => {
          if (room.roomId === result.room_id) {
            resolve(room.roomId);
            cleanUp();
          }
        },
        (e) => {
          logger.error("Failed to wait for the room to arrive", e);
        });
    };
    // snip
```

Then `roomId` can be used instead of `result.room_id` later on.